spec/index.adoc

```diff
@@ -1654,6 +1654,40 @@ There is no restriction on who can invoke this method.
 
 This method takes no input and returns 32 pseudo-random bytes to the caller. The return value is unknown to any part of the IC at time of the submission of this call. A new return value is generated for each call to this method.
 
+[#ic-http_request]
+=== IC method `http_request`
+
+This method makes an HTTP request to a given URL and returns the HTTP response, possibly after a transformation.
+
+The request must be _idempotent_, meaning that it must not change the state at the remote server, or the remote server has the means to identify duplicated requests.
```

Comment:
The canister has no means to ensure the request is idempotent, should we soften it to: 'the canister should aim to issue idempotent requests since otherwise the risk of failure increases'.